packages/perf/lib/index.js

```diff
@@ -47,12 +50,40 @@ class FirebasePerfModule extends FirebaseModule {
   constructor(...args) {
     super(...args);
     this._isPerformanceCollectionEnabled = this.native.isPerformanceCollectionEnabled;
+    this._instrumentationEnabled = this.native.isInstrumentationEnabled;
   }
 
   get isPerformanceCollectionEnabled() {
     return this._isPerformanceCollectionEnabled;
   }
 
+  get instrumentationEnabled() {
+    return this._instrumentationEnabled;
+  }
+
+  set instrumentationEnabled(enabled) {
+    if (!isBoolean(enabled)) {
+      throw new Error("firebase.perf().instrumentationEnabled = 'enabled' must be a boolean.");
+    }
+    if (Platform.OS == 'ios') {
+      // We don't change for android as it cannot be set from code, it is set at gradle build time.
+      this._instrumentationEnabled = enabled;
+      // No need to await, as it only takes effect on the next app run.
+      this.native.instrumentationEnabled(enabled);
+    }
+  }
+
+  get dataCollectionEnabled() {
+    return this._isPerformanceCollectionEnabled;
+  }
+
+  set dataCollectionEnabled(enabled) {
+    if (!isBoolean(enabled)) {
+      throw new Error("firebase.perf().dataCollectionEnabled = 'enabled' must be a boolean.");
+    }
+    this.setPerformanceCollectionEnabled(enabled);
+  }
+
```

Comment:
for 100% backwards compatibility, people that currently call setPerformanceCollectionEnabled on iOS were historically getting both the effect of what is now `this.native.setPerformanceCollectionEnabled` and `this.native.setInstrumentationEnabled` so this method `setPerformanceCollectionEnabled` just below (if you expand the diff) - which is non-JS-SDK / historical should also toggle isInstrumentationEnabled / call through to same on native (for iOS at least?) and in index.d.ts should have a note I think?

perfect JS compatibility (or as close as we can get) + backwards compatibility is super fun

Reply:
I think the setPerformanceCollectionEnabled method should also be marked deprecated ?